src/nodes/grpcNode.js

```diff
@@ -0,0 +1,137 @@
+const { ProcessStatus, Nodes } = require("@flowbuild/engine");
+const Ajv = require("ajv");
+const addFormats = require("ajv-formats");
+const { logger } = require("../utils/logger");
+const grpc = require("@grpc/grpc-js");
+const protoLoader = require("@grpc/proto-loader");
+const grpcReflection = require("grpc-reflection-js");
+const promiseWrapper = require("grpc-client-promise-wrapper");
+
+class GrpcNode extends Nodes.SystemTaskNode {
+  static get schema() {
+    return {
+      type: "object",
+      required: ["id", "name", "next", "type", "lane_id", "parameters"],
+      properties: {
+        id: { type: "string" },
+        name: { type: "string" },
+        next: { type: "string" },
+        type: { type: "string" },
+        category: { type: "string" },
+        lane_id: { type: "string" },
+        parameters: {
+          type: "object",
+          required: ["input"],
+          properties: {
+            input: {
+              type: "object",
+              required: ["server", "payload", "service", "method"],
+              properties: {
+                server: {
+                  oneOf: [{ type: "string" }, { type: "object" }],
+                },
+                service: {
+                  oneOf: [{ type: "string" }, { type: "object" }],
+                },
+                method: {
+                  oneOf: [{ type: "string" }, { type: "object" }],
+                },
+                payload: {
+                  oneOf: [{ type: "string" }, { type: "object" }],
+                },
+                descriptor: { type: "object" },
+                useReflection: { type: "boolean" },
+              },
+            },
+          },
+        },
+      },
+    };
+  }
+
+  static validate(spec, schema = null) {
+    const ajv = new Ajv({ allErrors: true });
+    addFormats(ajv);
+    const validationSchema = schema || GrpcNode.schema;
+    const validate = ajv.compile(validationSchema);
+    const validation = validate(spec);
+    return [validation, JSON.stringify(validate.errors)];
+  }
+
+  validate() {
+    return GrpcNode.validate(this._spec);
+  }
+
```

Comment:
Pelo que entendi (posso estar errado) se o `userReflection` for `false` o `descriptor` é obrigatório, correto? Caso seja isso que entendi, não precisaria de uma validação a mais pra esse caso? (Na validação do schema do nó ou do `executionData`?

Reply:
vc tem razão. Eu de início pensei com colocar um schema condicional, com o padrão do JSON-Schema é ser condicional a existência da propriedade - para fazer por valor seria necessário um if then else no schema.
Outra alternativa seria fazer invertendo a propriedade (useDescriptor), que nesse caso eu usaria o dependentRequired. Optei por não usar porque acredito que o padrão a priori seria usar descriptors.

Entendo que não é um problema ENORME porque sem o descriptor o nó simplesmente dá erro, mas reconheço que é um ponto de melhoria.